src/Rector/ClassMethod/ParamConverterAttributeToMapEntityAttributeRector.php

```diff
@@ -0,0 +1,197 @@
+namespace Rector\Symfony\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Attribute;
+use PhpParser\Node\AttributeGroup;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Use_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Php80\NodeAnalyzer\PhpAttributeAnalyzer;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://symfony.com/blog/new-in-symfony-6-2-built-in-cache-security-template-and-doctrine-attributes
+ *
+ * @see \Rector\Symfony\Tests\Rector\ClassMethod\ParamConverterAttributeToMapEntityAttributeRector\ParamConverterAttributeToMapEntityAttributeRectorTest
+ */
+final class ParamConverterAttributeToMapEntityAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const PARAM_CONVERTER_CLASS = 'Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter';
+
+    private const MAP_ENTITY_CLASS = 'Symfony\Bridge\Doctrine\Attribute\MapEntity';
+
+    private ?Use_ $use = null;
+
+    public function __construct(
+        private readonly PhpAttributeAnalyzer $phpAttributeAnalyzer,
+    ) {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace ParamConverter attribute with mappings with the MapEntity attribute',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeController
+{
+    #[Route('/blog/{date}/{slug}/comments/{comment_slug}')]
+    #[ParamConverter('post', options: ['mapping' => ['date' => 'date', 'slug' => 'slug']])]
+    #[ParamConverter('comment', options: ['mapping' => ['comment_slug' => 'slug']])]
+    public function showComment(
+        Post $post,
+        Comment $comment
+    ) {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeController
+{
+    #[Route('/blog/{date}/{slug}/comments/{comment_slug}')]
+    public function showComment(
+        #[\Symfony\Bridge\Doctrine\Attribute\MapEntity(mapping: ['date' => 'date', 'slug' => 'slug'])] Post $post,
+        #[\Symfony\Bridge\Doctrine\Attribute\MapEntity(mapping: ['comment_slug' => 'slug'])] Comment $comment
+    ) {
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Use_::class, ClassMethod::class];
```

Comment:
This rule should only handle the `ClassMethod`. What is the `Use_` here for?

Reply:
It removes the `use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;` when it is no longer needed.

Or should a rule not be responsible for cleaning unused _use_ statements?

Reply:
> Or should a rule not be responsible for cleaning unused use statements?

I see. That's handled somewhere in post-rectors, so you don't have to worry about it in every rule.

See https://github.com/rectorphp/rector-src/pull/3358

Reply:
I deleted the logic for the removal, however my tests now fail because the _use_ is still there.

Do I update the tests and leave them in?